Tutorials/WinForms/Classic/create-persistent-classes-and-connect-xpo-to-database.md

```diff
@@ -0,0 +1,99 @@
+# Create persistent classes and connect XPO to the database
+
+&nbsp;&nbsp;&nbsp;
+[Back to TOC](../../)
+&nbsp;&nbsp;&nbsp;
+[Step 2](/connect-data-grid-to-xpo-objects.md)   
+
+* Create a new Windows Forms App project in Visual Studio.
+* Add references to these assemblies ([Manage references in a project](https://docs.microsoft.com/en-us/visualstudio/ide/managing-references-in-a-project)):  
+  **DevExpress.Data**  
+  **DevExpress.Xpo**  
+  These assemblies are available under the **Assemblies > Extensions** category in the **Reference Manager**. 
+* Create the **DataAccess** folder and add the `Customer` class to it. This class is a model for the Customer table in a database.
+* Change the base class to `XPObject` and add a constructor with a `session` parameter:  
+    ``` csharp
+    using DevExpress.Xpo;
+
+    namespace DxSample.DataAccess {
+        public class Customer :XPObject {
+            public Customer(Session session) : base(session) { }
+        }
+    }
+    ```
+    **See also:**   
+    [XPO Classes Comparison](https://docs.devexpress.com/XPO/3311/concepts/xpo-classes-comparison)
+* Add the `FirstName` and `LastName` properties to the `Customer` class. XPO maps these proeprties to columns in the Customer table.
+    ```csharp
+    public string FirstName {
+        get { return fFirstName; }
+        set { SetPropertyValue(nameof(FirstName), ref fFirstName, value); }
+    }
+
+    private string fLastName;
+    public string LastName {
+        get { return fLastName; }
+        set { SetPropertyValue(nameof(LastName), ref fLastName, value); }
+    }
+    ```
+    The `SetPropertyValue` method raises the `PropertyChanged` event that is used by data bound controls to update their display text.
+* Add the read-only `ContactName` property. This property is not mapped to any column. Use the `FirstName` and `LastName` properties to calcualate the `ContactName` property value.
+    ```csharp
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Add the [PersistentAliasAttribute](https://docs.devexpress.com/XPO/DevExpress.Xpo.PersistentAliasAttribute) to the `ContactName` property. With this attribute, XPO will be able to include this property in a filter. The expression passed to the attribute constructor should follow the [Criteria Language Syntax](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax). XPO uses the Persistent Alias expression to build a SQL command compatible with the target database.
+    ```csharp
+    [PersistentAlias("concat(FirstName, ' ', LastName)")]
+    public string ContactName {
+        get { return string.Concat(FirstName, " ", LastName); }
+    }
+    ```
+* Use the same approach to add the `Order` class with the `ProductName`(String), `OrderDate`(DateTime), and `Freight`(decimal) properties. Check our example code in case of any difficulty:  
```

Comment:
"in case of any difficulty"?

Reply:
068d1827c0d6d6d5aac4d7f244e578f6d511cb11